x-pack/platform/packages/shared/kbn-streamlang/src/actions/action_metadata.ts

```diff
@@ -0,0 +1,362 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { i18n } from '@kbn/i18n';
+
+export interface ActionExample {
+  description: string;
+  yaml: string;
+}
+
+export interface ActionMetadata {
+  name: string;
+  description: string;
+  usage: string;
+  examples: ActionExample[];
+  tips?: string[];
+}
+
+export const ACTION_METADATA_MAP: Record<string, ActionMetadata> = {
```

Comment:
How do we keep these up to date? It's already missing `drop_document` and `replace`, I guess that will stay a theme.

<img width="289" height="326" alt="Image" src="https://github.com/user-attachments/assets/f5541123-8fbf-4043-a853-c3d8b6133d05" />

Maybe we can do a switch exhaustiveness check or something like that. Ideally it's all centralized in a manually managed OAS (cc @tonyghiani )